### DISCUSSION TITLE
Need some help understand versionning

### DISCUSSION THREAD

**Discussion by Xwilarg** [community/community#25956] [Actions]
<p>Hi,<br>
When you go create a GitHub package, there is a small text explaining how to create version, a part of it is saying:</p>
<blockquote>
<p>If the tag isn’t meant for production use, add a pre-release version after the version name. Some good pre-release versions might be v0.2-alpha or v5.9-beta.3.</p>
</blockquote>
<p>Then there is a link to <a href="https://semver.org/" rel="nofollow noopener">https://semver.org/</a> for people who want to read more<br>
But on that link, at the element 9, it’s written the following:</p>
<blockquote>
<p>A pre-release version MAY be denoted by appending a hyphen and a series of dot separated identifiers immediately following the patch version. […] Identifiers MUST NOT be empty. Numeric identifiers MUST NOT include leading zeroes.</p>
</blockquote>
<p>But the example given by GitHub (v0.2-alpha) have a leading 0 in it…<br>
So I just wanted to know if I understood something and/or which way was the “correct” one</p>
<p>Thanks in advance,</p>

**Comment by jcansdale** [comment, marked as answer]
<p>Hi <a class="mention" href="/Xwilarg">@Xwilarg</a>,</p>
<aside class="quote no-group" data-username="Xwilarg" data-post="1" data-topic="122233">
<div class="title">
<div class="quote-controls"></div>
 Xwilarg:</div>
<blockquote>
<p>But the example given by GitHub (v0.2-alpha) have a leading 0 in it…<br>
So I just wanted to know if I understood something and/or which way was the “correct” one</p>
</blockquote>
</aside>
<p>The way I read it is, the dot separated identifiers immediately following the patch version MUST NOT include leading zeroes. For example, you should avoid <code>v5.9-beta.007</code>. I don’t know if, <code>v0.2-alpha-0.2</code> would count as containing a leading zeros? I suspect they are okay because they don’t contain extra zeros.</p>

**Comment by Xwilarg** [comment]
<p>I guess I misunderstood that part, thanks for your answer!</p>
